shellwhat/run_file.py

```diff
@@ -0,0 +1,50 @@
+import subprocess
+
+from shellwhat.reporter import Reporter
+
+
+def run(state):
+    """Run the student file focused by ``check_file``.
+
+    Args:
+        state (State): state as passed by the SCT chain. Don't specify this explicitly.
+
+    :Example:
+
+        Suppose the student has a file ``script.sh`` in ``/home/repl/``::
+
+            echo 'test'
+
+        We can check if the file runs with this SCT::
+
+            Ex().check_file(
+                "script.sh"
+            ).run()
+    """
+    student_result, error = run_file(state.path)
+    return state.to_child(
+        student_result=student_result,
+        reporter=Reporter(state.reporter, errors=[error] if error else []),
+    )
+
+
+def run_file(path):
+    output, exc = None, None
+    try:
+        output = subprocess.check_output(path.__str__())
+    except subprocess.CalledProcessError as e:
+        exc = (
+            "returned non-zero exit status "
+            + str(e.returncode)
+            + " "
+            + str(e.output, "utf-8")
+        )
+    except PermissionError as e:
+        exc = "file not executable"
+
+    try:
+        output = str(output, "utf-8")
+    except TypeError:
+        pass
```

Comment:
When will this happen and what will happen next?

Reply:
This will happen if the output that the user made his script give is not utf-8. Or if output is empty.